src/bloqade/squin/rewrite/qasm2/core_to_squin.py

```diff
@@ -0,0 +1,38 @@
+from kirin import ir
+from kirin.dialects import py, func
+from kirin.rewrite.abc import RewriteRule, RewriteResult
+
+from bloqade import squin
+from bloqade.qasm2.dialects.core import stmts as core_stmts
+
+CORE_TO_SQUIN_MAP = {
+    core_stmts.QRegNew: squin.qubit.qalloc,
+    core_stmts.Reset: squin.qubit.reset,
+}
+
+
+class QASM2CoreToSquin(RewriteRule):
+
+    def rewrite_Statement(self, node: ir.Statement) -> RewriteResult:
+
+        if isinstance(node, core_stmts.QRegGet):
+            py_get_item = py.GetItem(
+                obj=node.reg,
+                index=node.idx,
+            )
+            node.replace_by(py_get_item)
+            return RewriteResult(has_done_something=True)
+
+        if isinstance(node, core_stmts.QRegNew):
+            args = (node.n_qubits,)
+        elif isinstance(node, core_stmts.Reset):
+            args = (node.qarg,)
+        else:
+            return RewriteResult()
+
+        new_stmt = func.Invoke(
+            callee=CORE_TO_SQUIN_MAP[type(node)],
```

Comment:
General question: Why do you rewrite to stdlib calls rather than statements?

Reply:
In my head it felt nicer to default to the stdlib versions because the output to the user would be on par with what they would see if they constructed the squin kernel through the interface we provide. Figured it makes debugging a bit nicer.

That being said, I don't have a strong preference it _has_ to be that way and if it ends up being the case we prefer the "unrolled"/flat form as the output I can easily do that (: